File: TP2-CPP/Kode Program TP 2/Televisi.cpp
#include <iostream>
#include <string>
#include "Elektronik.cpp" 

// Deklarasi class Televisi yang merupakan turunan (subclass) dari class Elektronik
class Televisi : public Elektronik {
protected:
    std::string ukuran;
    int stok;
    int tahun_produksi;

public:
    // Constructor dengan parameter lengkap untuk menginisialisasi atribut dari Produk, Elektronik, dan Televisi
    Televisi(std::string id_produk, std::string asal_produk, double harga, std::string lokasi_penyimpanan, std::string nama, std::string merk, std::string ukuran, int stok, int tahun_produksi)
        : Elektronik(id_produk, asal_produk, harga, lokasi_penyimpanan, nama, merk) {
        this->ukuran = ukuran;
        this->stok = stok;
        this->tahun_produksi = tahun_produksi;
    }

    // Getter methods untuk mendapatkan nilai
    std::string getUkuran() const {
        return ukuran;
    }

    int getStok() const {
        return stok;
    }

    int getTahunProduksi() const {
        return tahun_produksi;
    }

};

File: TP2-PHP/Kode Program/Televisi.php
<?php
require_once 'Elektronik.php';

class Televisi extends Elektronik {
    protected $ukuran;
    protected $stok;
    protected $tahun_produksi;
    protected $gambar;  

    public function __construct($id_produk, $asal_produk, $harga, $lokasi_penyimpanan, $nama, $merk, $ukuran, $stok, $tahun_produksi  ,$gambar) {
        parent::__construct($id_produk, $asal_produk, $harga, $lokasi_penyimpanan, $nama, $merk);
        $this->ukuran = $ukuran;
        $this->stok = $stok;
        $this->tahun_produksi = $tahun_produksi;
        $this->gambar = $gambar;
    }

    //getter methods
    public function getUkuran() {
        return $this->ukuran;
    }

    public function getStok() {
        return $this->stok;
    }

    public function getTahunProduksi() {
        return $this->tahun_produksi;
    }

    public function getGambar() {
        return $this->gambar;
    }

    
}
